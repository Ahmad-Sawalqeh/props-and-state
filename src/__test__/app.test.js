/* eslint-disable no-unused-vars */
import React from 'react';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter(), });

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../app.js';
import Form from '../components/form/form.js';

describe('<App />', () => {

  it('exists at the start of the application', () => {
    let app = shallow(<App />);
    expect(app.find('div').exists()).toBeTruthy();
  });

  // it('renders correctly', () => {
  //   const tree = renderer.create(<App />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  it('can render Form',()=>{
    let app = mount(<Form/>);
    let form = app.find('form');
    expect(form.exists()).toBeTruthy();
  });

  it('can render to the DOM',()=>{
    const rendered = renderer.create(<Form />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

});
