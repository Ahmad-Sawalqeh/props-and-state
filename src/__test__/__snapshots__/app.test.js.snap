// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> can render to the DOM 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="address"
    onChange={[Function]}
    placeholder="API url link ..."
    type="text"
  />
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
      value="get"
    >
      Get
    </button>
    <button
      onClick={[Function]}
      value="post"
    >
      Post
    </button>
    <button
      onClick={[Function]}
      value="put"
    >
      Put
    </button>
    <button
      onClick={[Function]}
      value="patch"
    >
      Patch
    </button>
    <button
      onClick={[Function]}
      value="delete"
    >
      Delete
    </button>
    <button
      className="go"
    >
      Go!
    </button>
  </div>
</form>
`;
